libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/TriggerConditions/OnCondition.cs

```diff
@@ -60,6 +68,35 @@ public OnCondition(string condition = null, List<Dialog> actions = null, [Caller
         [JsonIgnore]
         public virtual SourceRange Source => DebugSupport.SourceMap.TryGetValue(this, out var range) ? range : null;
 
+        /// <summary>
+        /// Gets or sets the rule priority expression where 0 is the highest and less than 0 is ignored.
+        /// </summary>
+        /// <value>Priority of condition expression.</value>
+        [JsonProperty("priority")]
+        public string Priority
+        {
+            get => priorityString;
+            set
+            {
+                priorityExpression = null;
+                priorityString = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether rule should only run once per unique set of memory paths.
+        /// </summary>
+        /// <value>Boolean if should run once per unique values.</value>
+        [JsonProperty("runOnce")]
+        public bool RunOnce { get; set; }
+
+        /// <summary>
+        /// Gets or sets the value of the Unique id for this condition.
+        /// </summary>
+        /// <value>Id for condition.</value>
+        [JsonIgnore]
+        public uint Id { get; set; }
```

Comment:
Hmm...all dialogs/actions/recognizers have an Id as a string.  can we make this a string for consistency ?